sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -2811,7 +2791,7 @@ lsc_slm_atomic_update(__ESIMD_NS::simd<uint32_t, N> offsets,
 /// @tparam Op is operation type.
 /// @tparam T is element type.
 /// @tparam N is the number of channels (platform dependent).
-/// @tparam DS is the data size.
+/// @tparam DS is the data size (unused/obsolete).
```

Comment:
This place looks concerning to me. Telling (unused/obsolete) for block_load/store was reasonable, because those did not support u16u32 conversions and the old lsc_atomic_update() does support it.
We need to be pretty subtle in this place and implementation of __ESIMD_DNS::atomic_update_impl. That impl function should support the DS template argument and support such u16u32.

The new API that calls atomic_update_impl should simply pass data_size::detault to impl call.

Reply:
You are right, sorry, I restored it.